saos-import/src/main/java/pl/edu/icm/saos/importer/notapi/common/content/ContentDownloadStepExecutionListener.java

```diff
@@ -0,0 +1,96 @@
+package pl.edu.icm.saos.importer.notapi.common.content;
+
+import java.io.File;
+import java.util.Collection;
+
+import org.springframework.batch.core.ExitStatus;
+import org.springframework.batch.core.StepExecution;
+import org.springframework.batch.core.StepExecutionListener;
+import org.springframework.beans.factory.annotation.Autowired;
+
+import pl.edu.icm.saos.importer.notapi.common.ImportFileUtils;
+import pl.edu.icm.saos.persistence.model.importer.RawSourceJudgment;
+
+import com.google.common.collect.Lists;
+
+/**
+ * Listener responsible for copying content files.
+ * 
+ * @author madryk
+ */
+public class ContentDownloadStepExecutionListener implements StepExecutionListener {
+    
+    private ImportFileUtils importFileUtils;
+    
+    private ImportContentFileUtils importContentFileUtils;
+    
+    private ContentSourceFileFinder importContentFileFinder;
+    
+    private String importDir;
+    
+    private String importContentDir;
+    
+    private Class<? extends RawSourceJudgment> rawJudgmentClass;
+    
+    
+    //------------------------ LOGIC --------------------------
+    
+    @Override
+    public void beforeStep(StepExecution stepExecution) {
+        
+        importContentFileUtils.deleteAllContentFiles(rawJudgmentClass);
+        
+        File contentSourceDir = new File(importContentDir);
+        
+        Collection<File> importFiles = importFileUtils.listImportFiles(importDir);
+        Collection<File> contentFiles = Lists.newArrayList();
+        
+        
+        for (File importFile : importFiles) {
+            File contentFile = importContentFileFinder.findContentFile(contentSourceDir, importFile);
+            contentFiles.add(contentFile);
+        }
+        
+        
+        for (File contentFile : contentFiles) {
+            importContentFileUtils.copyContentFile(contentFile, rawJudgmentClass);
```

Comment:
It would be clearer to use dir specified instead of rawJudgmentClass. And you could use here the standard FileUtils.copyFile(destDir, file)